openweathermap-ios/openweathermap-ios/BookmarkLocation/Model/BookmarkLocation.swift

```diff
@@ -0,0 +1,35 @@
+//
+//  BookmarkLocation.swift
+//  openweathermap-ios
+//
+//  Created by Cassio Sousa on 24/04/20.
+//
+
+import Foundation
+
+private enum BookmarkLocationKeys: String {
+    case title, latitude, longitude
+}
+
+final class BookmarkLocation: NSObject, NSCoding {
```

Comment:
Why do you need to inherit from `NSObject` again?

Reply:
Henrique, I tried to remove NSObject from class BookmarkLocation, but when do this I receive this error **does not implement methodSignatureForSelector: -- trouble ahead** when try execute NSKeyedArchiver.archivedData.

Reply:
Why don't you persist a simple Dictionary instead of archive an object?

Reply:
Well, I prefer use objects I think more readability but I made a test with removing NSObject
First I removed NSObject `final class BookmarkLocation: NSCoding {`
Next I tried to save the bookmark location ` defaults.set( ["id":bookmarkLocation], forKey: Bookmark.location.rawValue)`
Finally I get error
```
failed: caught "NSInvalidArgumentException", "Attempt to insert non-property list object {
    id = "openweathermap_iosTests.BookmarkLocation";
```


Why Can't I make this way? Is there a reason in particular?

[References](https://giovaninppc.github.io/Custom-Objects-User-Defaults/) 
Thank you 🙂

Reply:
I already approved the PR because the code is ok. But I strongly recommend you to change the current approach. 

Consider using `Struct+Codable+JSONDecoder=Swift` instead of `NSCoding+NSOBject+NSKeyedArchiver=Objective-C`.

Here is an article to help you:
https://www.hackingwithswift.com/example-code/system/how-to-load-and-save-a-struct-in-userdefaults-using-codable

Fell free to refactor this in the next PR.

Reply:
@henrique-morbin-ifood , I'm going to change it in the next PR